aptos-move/framework/aptos-framework/sources/account.move

```diff
@@ -329,6 +334,52 @@ module aptos_framework::account {
         rotation_capability
     }
 
+    // Offer the account owner's signer capablity to another account
+    public entry fun offer_signer_capability_ed25519(account: &signer, signer_capability_sig_bytes: vector<u8>, account_public_key_bytes: vector<u8>, recipient_address: address) acquires Account {
+        let addr = signer::address_of(account);
+        assert!(exists_at(addr) && exists_at(recipient_address), error::not_found(EACCOUNT_DOES_NOT_EXIST));
+
+        let pubkey = ed25519::new_unvalidated_public_key_from_bytes(account_public_key_bytes);
+        let signer_capability_sig = ed25519::new_signature_from_bytes(signer_capability_sig_bytes);
+
+        // Get the current authentication key of the account and verify that it matches with `account_public_key_bytes`
+        let account_resource = borrow_global_mut<Account>(addr);
+        assert!(verify_authentication_key_matches_ed25519_public_key(account_resource.authentication_key, account_public_key_bytes), EWRONG_CURRENT_PUBLIC_KEY);
+
+        //  Construct a SignerCapabilityOfferProofChallenge struct
+        let signer_capability_offer_proof_challenge = SignerCapabilityOfferProofChallenge {
+            sequence_number: account_resource.sequence_number,
+            recipient_address,
+        };
+
+        // Verify a digital-signature-based capability that assures us this signer capability offer was intended by the account owner
+        assert!(ed25519::signature_verify_strict_t(&signer_capability_sig, &pubkey, signer_capability_offer_proof_challenge), EINVALID_PROOF_OF_KNOWLEDGE);
+
+        // Add the recipient's address in account owner's signer capability offer once we verify that this action is intended by the account owner
+        option::fill(&mut account_resource.signer_capability_offer.for, recipient_address);
+    }
+
+
+    // Accept signer capability from `offerer_address` if there's an existing signer capability offer to the account owner
+    public fun accept_signer_capability_ed25519(account: &signer, offerer_address: address) : SignerCapability acquires Account {
```

Comment:
So this is only meant to be called from a module or script? The signer cap needs to be stored somewhere and can't just be given directly to an account